### ISSUE TITLE
Spike: debug bcsc login issue in prod

### ISSUE THREAD

**Issue by MCatherine1994** [bcgov/nr-forests-access-management#1272]
**Describe the task**
Meet with Wesley to check the bcsc login issue in production

**Acceptance Criteria**
- [x] Start with a meeting with Wesley 

**Additional context**
- Some thoughts: removing gender attribute might caused this error

![Image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/af72a599-e45b-47a4-83f0-38bebe6814ee)

- Our api log actually returns the user's information 

![Image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/94439c9c-8045-4e15-a6c8-d6452722b03a)

**Comment by MCatherine1994** [comment]
Tried a few things:

- Tried to add username to the read/write attribute permission, but got pipeline error username is not a valid read/write attribute.
- Tried to remove the username attribute mapping from terraform, and it's auto added in the mapping, and no difference
- Tried to remove all attribute mapping and login again, got the same error
- Also chatted with Wesley, prod login using mobile app shouldn't be the reason of this error

Submitted a AWS support request through AWS Prod, https://support.console.aws.amazon.com/support/home?region=ca-central-1#/case/?displayId=171218031100513&language=en

**Comment by MCatherine1994** [comment]
Just attached the email from Alan as our note, it's the summary of our meeting



![Image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/8161ab01-f959-432a-8aed-7b429174e061)

**Comment by MCatherine1994** [comment]
Ian found out we had an error in our code while reviewing the PROD-BCSC config in terraform: 

We hardcoded the IDP name for PROD-BCSC long time ago, but forgot to update it to the real one Wesley shared with us recently. So when the user_info endpoint tries to verify the "aud" received from bcsc, it throws a 401, and the PROD-BCSC IDP name doesn't match what we have in the code. 

![Image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/4c1b8da3-c6bf-4574-9a15-e80043b5cb55)


![Image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/9291d5f8-e382-4cad-9b27-6abf14618095)

Fixed in the pr https://github.com/bcgov/nr-forests-access-management/pull/1288, Ian and Basil helped review, updated the hardcode PROD-BCSC IDP name in both Auth Lambda and BCSC router. Deployed to production in v1.14.0, and verified the bcsc prod login works now with Maria. 

Notified Wesley and the AWS support team, thanks so much for everyone's support!!
